File: backend/model/history.go
package model

import "time"

type OpenAIMessage struct {
	Role    string `json:"role"`
	Content string `json:"content"`
}

type TrainingHistory struct {
	ID               uint              `json:"id" gorm:"primaryKey"`
	UserID           uint              `json:"user_id"`
	TrainingType     string            `json:"training_type"`
	CustomTrainingID *uint             `json:"custom_training_id"`
	Title            string            `json:"title"`
	IsFavorite       bool              `json:"is_favorite"`
	LastMessage      string            `json:"last_message"`
	Messages         []TrainingMessage `json:"messages" gorm:"foreignKey:HistoryID"`
	CreatedAt        time.Time         `json:"created_at"`
	UpdatedAt        time.Time         `json:"updated_at"`
}

type TrainingMessage struct {
	ID        uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	HistoryID uint      `json:"history_id"`
	Role      string    `json:"role"`
	Content   string    `json:"content"`
	SortOrder int       `json:"sort_order"`
	CreatedAt time.Time `json:"created_at"`
}

func (TrainingMessage) TableName() string {
	return "training_messages"
}

type ListHistoryRequest struct {
	Page       int    `form:"page" binding:"required,min=1"`
	PageSize   int    `form:"pageSize" binding:"required,min=1,max=100"`
	Title      string `form:"title"`
	IsFavorite *bool  `form:"is_favorite"`
}

type ListHistoryResponse struct {
	Total int64             `json:"total"`
	Items []TrainingHistory `json:"items"`
}

type UpdateFavoriteRequest struct {
	IsFavorite bool `json:"is_favorite"`
}

type UpdateTitleRequest struct {
	Title string `json:"title" binding:"required"`
}

File: frontend/src/service/api/history.ts
import { request } from '../request';

export interface TrainingHistory {
  id: number;
  user_id: number;
  training_type: string;
  title: string;
  is_favorite: boolean;
  messages: string;
  created_at: string;
  updated_at: string;
}

export interface ListHistoryParams {
  page: number;
  pageSize: number;
  title?: string;
  is_favorite?: boolean;
}

export interface ListHistoryResponse {
  total: number;
  items: TrainingHistory[];
}

export function fetchHistoryList(params: ListHistoryParams) {
  return request<ListHistoryResponse>({
    url: '/api/histories',
    method: 'get',
    params
  });
}

export function fetchHistoryDetail(id: number) {
  return request<TrainingHistory>({
    url: `/api/histories/${id}`,
    method: 'get'
  });
}

export function fetchUpdateFavorite(id: number, is_favorite: boolean) {
  return request({
    url: `/api/histories/${id}/favorite`,
    method: 'put',
    data: { is_favorite }
  });
}

export function fetchUpdateHistoryTitle(id: number, title: string) {
  return request({
    url: `/api/histories/${id}/title`,
    method: 'put',
    data: { title }
  });
}
